gamespot.com shows a smaller video when entering full screen

jernoble opened the pull request:
#### 5b6026cbed5508f67d6f3305a2e42e47e1a7aad5
<pre>
gamespot.com shows a smaller video when entering full screen
<a href="https://bugs.webkit.org/show_bug.cgi?id=244256">https://bugs.webkit.org/show_bug.cgi?id=244256</a>
&lt;rdar://98754870&gt;

Reviewed by Eric Carlson.

When Gamespot.com videos enter fullscreen, they have a rule with a max-width: property
set on their fullscreen element, that prevents the element from being resized to occupy
the entire screen.

Adopt CSS from the current Fullscreen API specification for the fullscreen element,
including an explicit and !important min, max, and explicit width and height.

* LayoutTests/fullscreen/full-screen-min-max-width-height-expected.txt: Added.
* LayoutTests/fullscreen/full-screen-min-max-width-height.html: Added.
* LayoutTests/fullscreen/full-screen-test.js:
(waitFor):
* Source/WebCore/css/fullscreen.css:

Canonical link: <a href="https://commits.webkit.org/253790@main">https://commits.webkit.org/253790@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec52224a5330e0f5fbaf0afa8c72029ae826ecec

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87086 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31173 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/17935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/96060 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149667 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29538 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/91113 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92702 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/23853 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/73906 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/79219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/78837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/17935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27266 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/17935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/72635 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27210 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/17935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/25889 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2677 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28894 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/36806 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/75439 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28835 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/17935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/16702 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

eric-carlson reviewed (approved):
R=me

youssefsoli commented on LayoutTests/fullscreen/full-screen-test.js:
```diff
@@ -158,7 +158,25 @@ function waitForEventTestAndEnd(element, eventName, testFuncString)
+        }, { once: true });
+    });
+}
+
```
This will just override the `waitFor` function above and this doesn't seem to be used in this patch?

jernoble replied:
Ah! I missed that I'd already imported `waitFor()` from video-test.js.

jernoble committed:
gamespot.com shows a smaller video when entering full screen
https://bugs.webkit.org/show_bug.cgi?id=244256
<rdar://98754870>

Reviewed by Eric Carlson.

When Gamespot.com videos enter fullscreen, they have a rule with a max-width: property
set on their fullscreen element, that prevents the element from being resized to occupy
the entire screen.

Adopt CSS from the current Fullscreen API specification for the fullscreen element,
including an explicit and !important min, max, and explicit width and height.

* LayoutTests/fullscreen/full-screen-min-max-width-height-expected.txt: Added.
* LayoutTests/fullscreen/full-screen-min-max-width-height.html: Added.
* LayoutTests/fullscreen/full-screen-test.js:
(waitFor):
* Source/WebCore/css/fullscreen.css:

Canonical link: https://commits.webkit.org/253790@main
